test/functional/Jsr292/build.xml

```diff
@@ -189,25 +200,32 @@
 				</javac>
 			</then>
 			<else>
-				<javac srcdir="${src}" destdir="${build}" debug="true" fork="true" executable="${compiler.javac}" includeAntRuntime="false" encoding="ISO-8859-1">
-					<src path="${src}" />
-					<src path="${transformerListener}" />
-					<src path="${TestUtilities}" /> 
-					<!-- exclude non test files for indyn test -->
-					<exclude name="**/indyn/BootstrapMethods.java" />
-					<exclude name="**/indyn/ComplexIndyGenerator.java" />
-					<exclude name="**/indyn/Helper.java" />
-					<exclude name="**/indyn/SimpleIndyGenerator.java" />
-					<exclude name="**/CrossPackageHelper.java"/>
-					<exclude name="**/attachAPI/**" />
-					<classpath>
-						<pathelement location="${bootstrapBuild}"/>
-						<pathelement location="${LIB_DIR}/asm-all.jar" />
-						<pathelement location="${LIB_DIR}/junit4.jar" />
-						<pathelement location="${LIB_DIR}/testng.jar"/>
-						<pathelement location="${LIB_DIR}/jcommander.jar"/>
-					</classpath>
-				</javac>
+				<property name="srcpath" location="${src}:${transformerListener}:${TestUtilities}" />
+				<!-- exclude non test files for indyn test -->
+				<property name="commonExcludes" value="**/indyn/BootstrapMethods.java,**/indyn/ComplexIndyGenerator.java,**/indyn/Helper.java,**/indyn/SimpleIndyGenerator.java,**/CrossPackageHelper.java,**/attachAPI/**" />
+				<path id="build.cp">
+					<pathelement location="${bootstrapBuild}/"/>
+					<fileset dir="${LIB_DIR}/" includes="asm-all.jar"/>
+					<fileset dir="${LIB_DIR}/" includes="junit4.jar"/>
+					<fileset dir="${LIB_DIR}/" includes="testng.jar"/>
+					<fileset dir="${LIB_DIR}/" includes="jcommander.jar"/>
+				</path>
+				<if>
+					<matches string="${JDK_VERSION}" pattern="^(9|(1[0-8]))$$" />
+					<then>
+						<!-- Java 9-18 -->
+						<javac srcdir="${srcpath}" excludes="${commonExcludes}" destdir="${build}" debug="true" fork="true" executable="${compiler.javac}" includeAntRuntime="false" encoding="ISO-8859-1" classpathref="build.cp">
+						</javac>
+					</then>
+					<else>
+						<!-- Java 19+ (SecurityManager removed) -->
+						<javac srcdir="${srcpath}" excludes="${commonExcludes}" destdir="${build}" debug="true" fork="true" executable="${compiler.javac}" includeAntRuntime="false" encoding="ISO-8859-1" classpathref="build.cp">
+							<!-- exclude tests that depend on SecurityManager -->
+							<exclude name="com/ibm/j9/jsr292/MethodTypeTests_SM.java"/>
+							<exclude name="com/ibm/j9/jsr292/bootstrap/Test_CallerSensitive.java"/>
```

Comment:
What error for this test? assuming it compiles even the security manager APIs are removed.

Reply:
`Test_CallerSensitive` depends on `CallerSensitiveClass` which extends `SecurityManager`

Reply:
It doesn't appear that `SecurityManager` is required by `CallerSensitiveClass`, can the dependency be removed?

Reply:
Tried removing it and seems like it is ok